conference/src/main/java/com/kyriba/conference/management/api/ScheduleController.java

```diff
@@ -30,78 +37,74 @@
 @RequestMapping("/api/v1/schedule")
 public class ScheduleController
 {
+  private final ScheduleService scheduleService;
+
+
+  @Autowired
+  public ScheduleController(ScheduleService scheduleService)
+  {
+    this.scheduleService = scheduleService;
+  }
+
 
   @ApiOperation(value = "Show conference schedule")
   @GetMapping(produces = APPLICATION_JSON_UTF8_VALUE)
   ScheduleResponse showSchedule()
   {
-    return ScheduleResponse.builder().presentations(buildPresentations()).build();
+    List<PresentationResponse> presentations = stream(scheduleService.getSchedule().spliterator(), false)
```

Comment:
This logic should be moved to service. And why do you use spliterator?

Reply:
fixed for spliterator.
why mapping to API DTO should be in the service? in the current case DTO and domain are more independent according to your comment:
> Если продукт уже стартовал и DTO начали отличаться или же API публичный и хочется избежать влияния изменений DTO с бизнес уровня, то еще в контроллер добавляется преобразование DTO.

Reply:
Domain model should not directly be accessible from services, only DTOs. Additional transformation in the controller is needed only if you are using dedicated DTOs for more stable HTTP API.

Reply:
done